runtime/src/sledge_abi.c

```diff
@@ -1039,3 +1040,126 @@ sledge_abi__wasi_snapshot_preview1_sock_shutdown(__wasi_fd_t fd, uint32_t how)
 {
 	return wasi_unsupported_syscall(__func__);
 }
+
+/**
+ * @param key
+ * @param key_len
+ * @returns value_len at key or 0 if key not present
+ */
+EXPORT uint32_t
+sledge_abi__scratch_storage_get_size(uint32_t key_offset, uint32_t key_len)
+{
+	struct sandbox *sandbox = current_sandbox_get();
+
+	sandbox_syscall(sandbox);
+
+	uint8_t *key = (uint8_t *)get_memory_ptr_for_runtime(key_offset, key_len);
```

Comment:
Can't this error out if they pass an offset beyond linear memory bounds? I'd imagine I'm wrong, and instead you just panic out of this specific sandbox in that case (`setjmp`).